### DISCUSSION TITLE
Get parameters of weight and GDN

### DISCUSSION THREAD

**Discussion by ElonGUI** [tensorflow/compression#158] [Q&A] [answered]
Hi,I am a hardware engineer.
It is an excellent project and I try to  deploy it in hardware. I want to get the model parameters of bmshj2018, but I can't find the paramet files belong to bmshj2018. The strings is unreadable after printing in tfci.py.   So what should I do to get parameters of weight and GDN? Thank you very much.

**Comment by jonaballe** [comment, marked as answer]
Hi, there are two options: `bmshj2018.py` only has source code, but no pretrained parameters. All the trained models we provide are accessible through `tfci.py`. They are not in a format that you can load into `bmshj2018.py`, but frozen as a TF .metagraph file. You can either extract the weights from this graph, or train your own model using the code in `bmshj2018.py` and then use those.
